packages/vm/src/runBlock.ts

```diff
@@ -432,7 +432,12 @@ export async function rewardAccount(
  * Returns the encoded tx receipt.
  */
 export function encodeReceipt(tx: TypedTransaction, receipt: TxReceipt) {
-  const encoded = rlp.encode(Object.values(receipt))
+  const encoded = rlp.encode([
+    (receipt as PreByzantiumTxReceipt).stateRoot ?? (receipt as PostByzantiumTxReceipt).status,
+    receipt.gasUsed,
+    receipt.bitvector,
+    receipt.logs,
+  ])
```

Comment:
What is this change here?

Reply:
ah I wrote a comment about this line to narrate why but I must not have submitted it.

I was pretty surprised to find `rlp.encode(Object.values(receipt))` was working to rlp the receipt - `Object.values` converts the object's properties to an array and I guess we were lucky enough that the array order was exactly matching to the yellow paper specification (`[stateRoot, gasUsed, logsBloom, logs]`, post-Byzantium `stateRoot` replaced with `status`). I think being more explicit here is better for readability/understandability/upgradability so I made the update.